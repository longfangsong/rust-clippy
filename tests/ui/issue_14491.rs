//@aux-build:macro_rules.rs
#![deny(clippy::missing_panics_doc)]
#![deny(clippy::missing_errors_doc)]
#![deny(clippy::allow_attributes)]

use std::env::var;

pub fn f() -> Result<(), &'static str> {
    //~^ missing_errors_doc
    match "test" {
        "hello" => Ok(()),
        _ => Err("Oh no"),
    }
}

pub fn g() {
    //~^ missing_panics_doc
    panic!();
}

#[expect(clippy::missing_panics_doc)]
#[expect(clippy::missing_errors_doc)]
pub fn main() -> Result<(), &'static str> {
    let val = var("Hello").unwrap();
    match val.as_str() {
        "hello" => Ok(()),
        _ => Err("Oh no"),
    }
}
