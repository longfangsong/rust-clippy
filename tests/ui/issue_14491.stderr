error: docs for function returning `Result` missing `# Errors` section
  --> tests/ui/issue_14491.rs:8:1
   |
LL | pub fn f() -> Result<(), &'static str> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> tests/ui/issue_14491.rs:3:9
   |
LL | #![deny(clippy::missing_errors_doc)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: docs for function which may panic missing `# Panics` section
  --> tests/ui/issue_14491.rs:16:1
   |
LL | pub fn g() {
   | ^^^^^^^^^^
   |
note: first possible panic found here
  --> tests/ui/issue_14491.rs:18:5
   |
LL |     panic!();
   |     ^^^^^^^^
note: the lint level is defined here
  --> tests/ui/issue_14491.rs:2:9
   |
LL | #![deny(clippy::missing_panics_doc)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

